### PR TITLE
[ticket/14547] Add Vagrant support to phpBB

### PR DESCRIPTION
**Pull request by VSEphpbb**
https://tracker.phpbb.com/browse/PHPBB3-14547

This should be tested and evaluated first, to determine if it's a useful development addition/tool for phpBB to have. I made a pretty in depth documentation for it, the idea being to make coding, testing, debugging easier for contributors, especially the more casual contributors.

### CODE REVIEW

**Comment by VSEphpbb** [comment]
Damn, looks like the package I want to use requires PHP 5.5+

**Comment by marc1706** [comment]
You should be able to use `~2.2` for having PHP 5.4 compatibility.

**Comment by VSEphpbb** [comment]
Yeah, but the the point was to use the box with PHP 7. The 2.2 boxes are 5.6

**Comment by marc1706** [comment]
Maybe something for master then? @Nicofuma

**Comment by Nicofuma** [comment]
If we bump the php req in master yes (and I think we should). But we need to announce that first

**Comment by VSEphpbb** [comment]
You going to bump it all the way up to PHP 5.6?

**Comment by Nicofuma** [comment]
You have a conflict

**Comment by marc1706** [comment]
@VSEphpbb instead of merging 3.2.x into your branch, just rebase your branch onto 3.2.x next time:
```git rebase upstream/3.2.x```
Make sure to run `git fetch upstream` before that.

**Comment by VSEphpbb** [comment]
My only question in this PR to you dev team guys is about the backup strategy being used to backup config.php files. Since a new config.php is generated for the vagrant environment, it backups up any existing config.php files, so as not to blow out any settings a user has if they also use the board in local AMP stacks. 

The current strategy is to have it backed up during the shell scripts, which result in:
config.php.bak~1
config.php.bak~2
etc..

Hence the modification to the gitignore file.

File: phpBB/docs/vagrant.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+> Note: destroying the vagrant server will remove all traces of the VM from your computer, reclaiming any disk space used by it. However, it also means the next time you vagrant up, you will be creating a brand new VM with a fresh install of phpBB and a new database.
+
+## Customising the phpBB configuration
+
+By default, phpBB is pre-configured to install with a MySQL database. You can, however, switch to PostegreSQL or SQLite3 by editing the `phpbb-install-config.yml` file in the vagrant directory. The next time you run `vagrant up` (or `vagrant provision`) it will be installed under the new configuration.
+
+If you prefer to access phpBB from the more friendly URL `http://phpbb.app` then you must update your computer's hosts file. This file is typically located at `/etc/hosts` for Mac/Linux or `C:\Windows\System32\drivers\etc\hosts` for Windows. Open this file and add the following line to it, at the very bottom, and save.
+
+```
+192.168.10.10  phpbb.app
```
Comment (marc1706) [inline phpBB/docs/vagrant.md]: Seems like a different IP than above?

File: vagrant/after.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Update sqlite db file permissions
+sudo chown -R vagrant /tmp/phpbb.sqlite3
+
+# Add DEBUG mode to phpBB to remove annoying installer warnings
+sed -i "$ a @define('DEBUG', true);" ${PHPBB_CONFIG}
+
+# Update the PHP memory limits (enough to allow phpunit tests to run)
+sed -i "s/memory_limit = .*/memory_limit = 768M/" /etc/php5/fpm/php.ini
+
+echo "Your board is ready at http://192.168.10.10/"
```
Comment (marc1706) [inline vagrant/after.sh]: Also seems to be a different IP than mentioned at the beginning of the documentation.

File: vagrant/after.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+php ${PHPBB_PATH}/phpBB/install/phpbbcli.php install ${PHPBB_INSTALL}
+
+# Update sqlite db file permissions
+sudo chown -R vagrant /tmp/phpbb.sqlite3
+
+# Add DEBUG mode to phpBB to remove annoying installer warnings
+sed -i "$ a @define('DEBUG', true);" ${PHPBB_CONFIG}
+
+# Update the PHP memory limits (enough to allow phpunit tests to run)
+sed -i "s/memory_limit = .*/memory_limit = 768M/" /etc/php5/fpm/php.ini
```
Comment (marc1706) [inline vagrant/after.sh]: phpunit might actually grow above that so I'd recommend 1024M.

**Comment by marc1706** [comment]
Rest looks good, only the IP at the beginning seems to be wrong.

**Comment by VSEphpbb** [comment]
Updated

**Comment by VSEphpbb** [comment]
Failure is unrelated

**Pull request merged**